pandas/core/dtypes/concat.py

```diff
@@ -559,3 +562,43 @@ def convert_sparse(x, axis):
         # coerce to object if needed
         result = result.astype('object')
     return result
+
+
+def _concat_indexes_same_dtype_rangeindex(indexes):
```

Comment:
maybe `_concat_rangeindex_same_dtype` ? (little bit shorter and also clear I think)